app/gui/suggestion-database/src/entry.rs

```diff
@@ -888,19 +888,15 @@ fn resolve_tag_value<'a>(
     db: &SuggestionDatabase,
     parser: &parser::Parser,
 ) -> TagValueResolution<'a> {
-    let qualified_name = QualifiedName::from_text(raw_expression).ok();
-    if let Some(mut qualified_name) = qualified_name {
-        let entry = db.lookup_by_qualified_name(&qualified_name).or_else(move || {
-            // Second try at lookup. Enum values usually have the module and type name deduplicated,
-            // but the resolution database does not take that into account. So we need to manually
-            // expand the name and try again.
-            let last_segment = qualified_name.pop_segment()?;
-            let segment_to_duplicate = qualified_name.pop_segment()?;
-            qualified_name.push_segment(segment_to_duplicate.clone());
-            qualified_name.push_segment(segment_to_duplicate);
-            qualified_name.push_segment(last_segment);
-            db.lookup_by_qualified_name(&qualified_name)
-        });
+    let stripped_expr = raw_expression.trim_start_matches('(').trim_end_matches(')');
+    let qualified_name = QualifiedName::from_text(stripped_expr).ok();
+    if let Some(qualified_name) = qualified_name {
+        let entry = db.lookup_by_qualified_name(&qualified_name);
+
+        // If a lookup of a fully qualified name have failed, the name is likely not actually fully
+        // qualified. Try to resolve it as partially qualified, ignoring potential ambiguity. Only
+        // first match in the database is considered.
+        let entry = entry.or_else(|| db.find_public_entry_by_partial_name(stripped_expr));
```

Comment:
I don't fully understand the reexport field. What would be the desired logic there exactly? Should it be "module segments match either `defined_in` or `reexported_in`"? Is it possible to have many reexports of the same type/module in various places?

Reply:
> I don't fully understand the reexport field. What would be the desired logic there exactly? Should it be "module segments match either defined_in or reexported_in"? Is it possible to have many reexports of the same type/module in various places?

`reexport` field contains "least nested" reexport of the entity represented by the given entry.

Yes, you can have one type reexported in many places. As the field was mainly designed for the Searcher, it contains only the "main" reexport, which was assumed to be the shallowest in the given library.

The question is if the library author, making one reexport "a main one" would also like to refer to it in tags, or whatever description of drop-downs?

Reply:
In this case I'm going to say that we don't want to include reexports in this search logic. It  would make the lookup significantly more expensive. It is only meant as a fallback logic in cases that the FQN was not specified correctly. The tag values are supposed to always use FQNs already, and the expressions in dynamic dropdowns should migrate to that sooner rather than later.